docs/RelaxNGSupport.md

```diff
@@ -0,0 +1,102 @@
+# RelaxNG support
+
+[vscode-xml](https://github.com/redhat-developer/vscode-xml) provides validation, completion, hover [RelaxNG](https://relaxng.org/) support in XML file since `0.22.0`:
+
+![RelaxNG support](images/RelaxNG/RelaxNGSupport.gif)
+
+It supports both:
+
+ * [RelaxNG XML syntax](https://relaxng.org/tutorial-20011203.html).
+ * [RelaxNG Compact syntax](https://relaxng.org/compact-tutorial-20030326.html).
+
+![XML Validated with RelaxNG](images/RelaxNG/XMLValidatedWithRelaxNG.png)
+
+To bind your XML file with `RelaxNG grammar` (`file` or `URL`), you can do that :
+
+ * with `xml-model` processing instruction
+
+```xml
+<?xml-model href="addressBook.rng" type="application/xml"?>
+<addressBook>   
+  ...
+</addressBook> 
+```
+
+ * with `XML catalog` like you can do that [with XSD](Validation#XML-catalog-with-XSD)
+ * with `file association` like you can do that [with XSD](Validation#XML-file-association-with-xsd)
+
+It provides too support for the `RelaxNG grammar` for:
+
+ * [RelaxNG XML syntax](https://relaxng.org/tutorial-20011203.html): you will benefit the same support than XML, XSD (completion, hover, validation, etc).
+ 
+![RelaxNG XML Syntax Overview](images/RelaxNG/RelaxNGXMLSyntaxOverview.png)
+  
+ * [RelaxNG Compact syntax](https://relaxng.org/compact-tutorial-20030326.html) you will benefit only with syntax coloration.
+ 
+![RelaxNG Compact Syntax Overview](images/RelaxNG/RelaxNGCompactSyntaxOverview.png)
+
+# How to start?
+
+## Using snippets
+
+You can use snippets to create quickly :
+
+ * an XML file associate with a RelaxNG grammar
+ * a RelaxNG XML file
+ 
+Here a demo:
+
+![RelaxNG snippets support](images/RelaxNG/RelaxNGSnippetsSupport.gif)
+
+## Creating XML, rng, rnc files
+
+Create RelaxNG grammar with Compact syntax `addressBook.rnc` file like this: 
+
+```rnc
+element addressBook {
+  element card {
+    element name { text },
+    element email { text }
+  }*
+} 
+```
+
+Create RelaxNG grammar with XML syntax `addressBook.rng` file like this: 
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<grammar xmlns="http://relaxng.org/ns/structure/1.0">
+    <start>
+        <element name="addressBook">
+            <zeroOrMore>
+                <element name="card">
+                    <ref name="cardContent"/>
+                </element>
+            </zeroOrMore>
+        </element>
+    </start>
+    <define name="cardContent">
+        <element name="name">
+            <text/>
+        </element>
+        <element name="email">
+            <text/>
+        </element>
+    </define>
+</grammar>
+```
+
+Create the XML `addressBook.xml` file to validate with `addressBook.rng` like this: 

```

Comment:
The following file, `addressBook.xml`, references `addressBook.rng` using the `xml-model` processing instruction. vscode-xml will provide the features based on the grammar shown in the demo at the beginning of this page when you open `addressBook.xml`.

Reply:
fixed